### ISSUE TITLE
Save question doesn't work when using a mathematical operation in a custom expression

### ISSUE THREAD

**Issue by steinmann321** [metabase/metabase#10985]
Having a mathematical operation between two aggregated fields prevents from saving a question. 
Testet with Version 0.33.0 and 0.33.3

**Logs**
[localhost-1569406638153.log](https://github.com/metabase/metabase/files/3651905/localhost-1569406638153.log)

**To Reproduce**
1. Go to 'new question'
2. Select 'sample dataset' => 'products'
3. Add a custom expression 'Summe(PRICE) / Summe(RATING)'  => maybe Sum(PRICE) / Sum(RATING) in english?
4. Click 'Save': Nothing happens but the javascript error in shows up as posted in the logfile

**Expected behavior**
The 'Save new question' dialog should appear

**Screenshots**
![image](https://user-images.githubusercontent.com/10723267/65592668-4f951b00-df8f-11e9-8d0d-575dcde2eb5e.png)

**Information about your Metabase Installation:**
{
  "browser-info": {
    "language": "de-DE",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "10.0.2+13",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "10.0.2",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "10.0.2+13",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "de",
    "user.timezone": "Europe/Prague"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "run-mode": "prod",
    "version": {
      "date": "2019-09-20",
      "tag": "v0.33.3",
      "branch": "release-0.33.x",
      "hash": "86d32bf"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}

- Your browser and the version: Chrome 77.0.3865.90
- Your operating system: Windows 10.1809
- Your databases: MySQL
- Metabase version: 0.33.0 and 0.33.3
- Metabase hosting environment: Docker, Jar File
- Metabase internal database: MySQL, H2

**Severity**
Blocker

**Comment by flamber** [comment]
Hi @steinmann321 
This is generating the same problem as #10844, but this is in QB/Notebook. When there's no name, then a lot of the interface fails.
Work-around: Give the Custom Expression a name.

There's a lot of warnings/errors, when using Custom Expressions (even when it's working), so it can be tricky to find the right errors, but I think this is the one:
```
Uncaught Error: `query` is a required parameter to format expressions
    at formatFieldReference (formatter.js:61)
    at format (formatter.js:39)
    at formatter.js:83
    at Array.map (<anonymous>)
    at formatAggregation (formatter.js:83)
    at format (formatter.js:48)
    at formatter.js:75
    at Array.map (<anonymous>)
    at formatMath (formatter.js:74)
    at format (formatter.js:45)
```

**Comment by mazameli** [comment]
Maybe we should just require a label for custom expressions?

**Comment by steinmann321** [comment]
Hi @flamber, naming the metric did the trick, thank you!

**Comment by paulrosenzweig** [comment]
Fixed in #11065
